api/src/pcapi/core/subscription/dms/api.py

```diff
@@ -108,22 +141,20 @@ def handle_dms_application(
     if fraud_check is None:
         fraud_check = fraud_dms_api.create_fraud_check(user, application_number, application_content)
     else:
-        fraud_check_content = typing.cast(fraud_models.DMSContent, fraud_check.source_data())
-        if dms_application.latest_modification_datetime == fraud_check_content.get_latest_modification_datetime():
-            logger.info("[DMS] Application already processed", extra=log_extra_data)
+        if not _has_application_update(fraud_check, application_content):
+            logger.info("[DMS] FraudCheck already up to date", extra=log_extra_data)
             return fraud_check
 
+        _update_fraud_check_with_new_content(fraud_check, application_content)
+
     if fraud_check.status == fraud_models.FraudCheckStatus.OK:
         logger.warning("[DMS] Skipping because FraudCheck already has OK status", extra=log_extra_data)
         return fraud_check
 
-    fraud_check.resultContent = application_content.dict()
-    fraud_check.eligibilityType = fraud_api.decide_eligibility(
-        user, application_content.get_birth_date(), application_content.get_registration_datetime()
-    )
+    birth_date_error = _compute_birth_date_error_details(fraud_check, application_content)
 
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏On est pas vraiment sur de la refacto ici, on change un peu la logique
```
Avant:
Vérifier si on doit update:
    Non -> Retourner le fraud_check
    Oui -> continue
Vérifier si status OK
    Oui -> skip, déjà OK
    Non -> continue
Update le fraud check

Après:
Vérifier si on doit update:
    Non -> Retourner le fraud_check
    Oui -> **update le fraud check**
Vérifier si status OK
    Oui -> skip, déjà OK
    Non -> continue
```


J'ai l'impression qu'on fait l'update trop tôt et qu'on risaue de maj un fraud_check déjà OK

Reply:
c'est vrai, je corrige